Automattic/jetpack
# Jetpack AI Image: Register ai-general-purpose-image-generator beta extension

## Proposed changes:

* Create `ai-general-purpose-image-generator` beta flag to control the General Purpose Image generator release

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Open the dev console
* With the `beta` extensions disabled, confirm that `Jetpack_Editor_Initial_State.available_blocks["ai-general-purpose-image-generator"]` is not set:

```
Jetpack_Editor_Initial_State.available_blocks['ai-general-purpose-image-generator']
```

<img width="300" alt="Screenshot 2024-06-06 at 17 00 37" src="https://github.com/Automattic/jetpack/assets/6760046/70e50722-e703-4e76-a8e3-fb728d1f5d22">

* With the `beta` extensions enabled, confirm that `Jetpack_Editor_Initial_State.available_blocks["ai-general-purpose-image-generator"]` is set to `available: true`

<img width="300" alt="Screenshot 2024-06-06 at 17 01 15" src="https://github.com/Automattic/jetpack/assets/6760046/5811d6b8-635f-492f-87f4-21ac49c6d8c8">
